node_modules/oae-ui/lib/api.js

```diff
@@ -811,6 +813,38 @@ var _replaceOptimizedPaths = function(skinVariables) {
     return replacedVariables;
 };
 
+/**
+ * Uploads a logo file for a tenant.
+ *
+ * @param  {Context}    ctx             Current request context
+ * @param  {Object}     file            A file object as returned by express
+ * @param  {String}     tenantAlias     The alias of the tenant for which to upload the logo
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ * @param  {String}     callback.url    The signed URL for the new logo file
+ */
+var uploadLogoFile = module.exports.uploadLogoFile = function(ctx, file, tenantAlias, callback) {
+    tenantAlias = tenantAlias || ctx.tenant().alias;
+    var tenant = ctx.tenant();
+    if (!ctx.user() || !ctx.user().isAdmin(tenantAlias)) {
+        return callback({'code': 401, 'msg': 'Only administrators can upload new logos for tenants'});
+    }
+    var extension = file.name.split('.').pop();
+    if (!extension.match(/(gif|jpe?g|png)$/i)) {
+        return callback({'code': 500, 'msg': 'File has an invalid mime type'});
+    }
+
+    var options = {
+        'prefix': util.format('logos/%s', tenantAlias)
+    }
```

Comment:
I'd leave it for now since it's not something that's been changed in this PR (and I'm not sure about 2 either).